tests/integration/test_bigquery.py

```diff
@@ -1,48 +1,86 @@
+from typing import List
 import pandas as pd
+import pandas_gbq
+import pytest
+from numpy import ndarray
 
+from viadot.exceptions import CredentialError, DBDataAccessError
 from viadot.sources import BigQuery
 
-BIGQ = BigQuery(credentials_key="BIGQUERY_TESTS")
 
+@pytest.fixture(scope="function")
+def BIGQ():
+    BQ = BigQuery(credentials_key="BIGQUERY_TESTS")
+    yield BQ
 
-def test_list_project():
+
+@pytest.fixture(scope="session")
+def inset_into_tables():
+    table_id1 = "manigeo.manigeo_tab"
+    table_id2 = "manigeo.space"
+    df = pd.DataFrame({"my_value": ["val1", "val2", "val3"]})
+    pandas_gbq.to_gbq(df, table_id1, if_exists="replace")
+    pandas_gbq.to_gbq(df, table_id2, if_exists="replace")
+
+
+QUERY = """
+        SELECT name, SUM(number) AS total
+        FROM `bigquery-public-data.usa_names.usa_1910_2013`
+        GROUP BY name, gender
+        ORDER BY total DESC 
+        LIMIT 4
+        """
+
+
+def test_credentials():
+    with pytest.raises(CredentialError, match=r"Credentials not found."):
+        _ = BigQuery(credentials_key="BIGQUERY_TESTS_FAKE")
+
+
+def test_list_project(BIGQ):
     project = BIGQ.get_project_id()
+
     assert project == "manifest-geode-341308"
 
 
-def test_list_datasets():
+def test_list_datasets(BIGQ):
     datasets = list(BIGQ.list_datasets())
+
     assert datasets == ["manigeo", "official_empty"]
 
 
-def test_list_tables():
+def test_list_tables(BIGQ):
     datasets = BIGQ.list_datasets()
     tables = list(BIGQ.list_tables(datasets[0]))
-    assert tables == ["space", "test_data", "manigeo_tab"]
+    assert "space" and "manigeo_tab" in tables
```

Comment:
Previous assertion was more clear

Reply:
It depends on the Bigquery project structure. It was safer to create a test in this way because we are testing on 'test environment'.